compiler/passes/scopeResolve.cpp

```diff
@@ -1971,15 +2004,11 @@ bool lookupThisScopeAndUses(const char*           name,
   }
 
   if (Symbol* sym = inType(name, scope)) {
-    if (isRepeat(sym, symbols) == true) {
-      // If we're looking at the exact same Symbol, there's no need to add it
-      // and we can just return.
-      return true;
```

Comment:
It looks like you're removing a short-cut out of the function by doing this.  Wouldn't that offset your performance benefit from the other changes?

Reply:
Right, I wanted to remove the return since the end of the function does some stuff to handle the cache. But AFAIK this code can only ever run at a type scope vs module scope so that should not matter. I've added it back if `useCache == false`.